frontend/src/pages/ContractInteractionPage.js

```diff
@@ -0,0 +1,174 @@
+	const [nftHash, setNftHash] = useState("");
+	const [ownedNftIndex, setOwnedNftIndex] = useState("");
+
+	const [createDropResult, setCreateDropResult] = useState("");
+	const [joinDropResult, setJoinDropResult] = useState("");
+	const [dropDropResult, setDropDropResult] = useState("");
+	const [buyNftResult, setBuyNftResult] = useState("");
+	const [realNftIndex, setRealNftIndex] = useState("");
+
+
+
+	return <CenterFlexBox>
+		<RoundedEdgesButton style={{ backgroundColor: "blue" }} onClick={() => ethereumContractApi.init()}>Initialize connections</RoundedEdgesButton>
+		<RoundedEdgesButton style={{ backgroundColor: "blue" }} onClick={async () => setBalance(await ethereumContractApi.getBalanceOfSelectedAddress())}>Get Balance balance</RoundedEdgesButton>
+		{(balance/1000000000000000000).toFixed(9)}
+		<RoundedEdgesButton style={{ backgroundColor: "blue" }} onClick={async () => setAddress(await ethereumContractApi.getSelectedAddress())}>Get selected address</RoundedEdgesButton>
+		{address}
+		<RoundedEdgesButton style={{ backgroundColor: "blue" }} onClick={async () => setDropTime(JSON.stringify(await ethereumContractApi.getDropTime(parseInt(dropHash))))}>Get Droptime (enter drop hash below)</RoundedEdgesButton>
+		time: {dropTime}
+		<RoundedEdgesButton style={{ backgroundColor: "blue" }} onClick={async () => setMethods(await ethereumContractApi.getMethods())}>Get all methods of contract</RoundedEdgesButton>
+		<ul>
+			{methods.map((e, i) => <li key={i}>{e}</li>)}
+		</ul>
+		<TextField sx = {{ ...inputFieldStyle, paddingBottom: "20px" }} id="outlined-multiline-flexible" label="Drop hash number" value={dropHash} onChange={event => setDropHash(event.target.value)} />
+		<TextField sx = {{ ...inputFieldStyle, paddingBottom: "20px" }} id="outlined-multiline-flexible" label="Number of NFTs to join drop for" value={numberOfNFTsToBuy} onChange={event => setNumberOfNFTsToBuy(event.target.value)} />
+		<TextField sx = {{ ...inputFieldStyle, paddingBottom: "20px" }} id="outlined-multiline-flexible" label="Droptime" value={droptime} onChange={event => setDroptime(event.target.value)} />
+		<TextField sx = {{ ...inputFieldStyle, paddingBottom: "20px" }} id="outlined-multiline-flexible" label="Number of NFTs to create" value={numberOfNFTsToCreate} onChange={event => setNumberOfNFTsToCreate(event.target.value)} />
+		<TextField sx = {{ ...inputFieldStyle, paddingBottom: "20px" }} id="outlined-multiline-flexible" label="NFT hash to buy" value={nftHash} onChange={event => setNftHash(event.target.value)} />
+		<TextField sx = {{ ...inputFieldStyle, paddingBottom: "20px" }} id="outlined-multiline-flexible" label="Get index (0,1,2...) of owned NFT" value={ownedNftIndex} onChange={event => setOwnedNftIndex(event.target.value)} />
+
+		<ParagraphTypography>Create a new drop for the above specified amount of NFTs to create for a specified drop time.</ParagraphTypography>
+
+		<RoundedEdgesButton
+			style={{ backgroundColor: "blue" }} onClick={async () => {
```

Comment:
For example this inlined `onClick` function is a not good for the software-engineering principles that I learnt. Same with the other buttons below. But due to equal structure, this is less of a problem than it could be.

Reply:
Once I create individual components for the buttons, I will seperate them from the components.

Reply:
Alright, thank you :-) . As long as you can explain your reason well, I'm fine with that.